src/Dapr.AspNetCore/DaprEndpointRouteBuilderExtensions.cs

```diff
@@ -78,5 +77,67 @@ public static IEndpointConventionBuilder MapSubscribeHandler(this IEndpointRoute
                 await writer.FlushAsync();
             });
         }
+
+        /// <summary>
+        /// Maps an endpoint that will respond to requests to <c>/dapr/subscribe</c> from the
+        /// Dapr runtime.
+        /// </summary>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder" />.</param>
+        /// <param name="options">Configuration options</param>
+        /// <returns>The <see cref="IEndpointConventionBuilder" />.</returns>
+        /// <seealso cref="MapSubscribeHandler(IEndpointRouteBuilder)"/>
+        public static IEndpointConventionBuilder MapSubscribeHandler(this IEndpointRouteBuilder endpoints, SubscribeOptions options)
+        {
+            if (endpoints is null)
+            {
+                throw new System.ArgumentNullException(nameof(endpoints));
+            }
+
+            return endpoints.MapGet("dapr/subscribe", async context =>
+            {
+                var dataSource = context.RequestServices.GetRequiredService<EndpointDataSource>();
+                var entries = dataSource.Endpoints
+                    .OfType<RouteEndpoint>()
+                    .Where(e => e.Metadata.GetMetadata<ITopicMetadata>()?.Name != null)   // only endpoints which have  TopicAttribute with not null Name.
+                    .Distinct()
+                    .Select(e => (e.Metadata.GetMetadata<ITopicMetadata>().PubsubName, e.Metadata.GetMetadata<ITopicMetadata>().Name, e.Metadata.GetMetadata<IRawTopicMetadata>()?.EnableRawPayload, e.RoutePattern));
+
+
+                var logger = context.RequestServices.GetService<ILoggerFactory>().CreateLogger("DaprTopicSubscription");
+
+                ArrayList subscriptions = new();
+                foreach (var entry in entries)
+                {
+                    // only return topics which have routes without parameters.
+                    if (entry.RoutePattern.Parameters.Count > 0)
+                    {
+                        if (logger != null)
+                        {
+                            logger.LogError("Topic subscription doesn't support route with parameters. Subscription for topic {name} is removed.", entry.Name);
+                        }
+
+                        continue;
+                    }
+
+                    var route = string.Join("/",
+                        entry.RoutePattern.PathSegments
+                        .Select(segment => string.Concat(segment.Parts.Cast<RoutePatternLiteralPart>()
+                        .Select(part => part.Content))));
+
+                    subscriptions.Add(new
+                    {
+                        topic = entry.Name,
+                        pubsubName = entry.PubsubName,
+                        route,
+                        metadata = new
+                        {
+                            rawPayload = (entry.EnableRawPayload == true) ? "true" : (entry.EnableRawPayload == false) ? "false" : options.EnableRawPayload.ToString().ToLower()
```

Comment:
Is `rawPayload` supposed to be a string in the response or a boolean? It looks like this is going to serialize as: `{ "rawPayload": "true" }` instead of `{ "rawPayload": true }`.

You can probably simplify this to `rawPayload = entry.EnableRawPayload ?? false`

Reply:
This is supposed to be string and not boolean.
https://docs.dapr.io/developing-applications/building-blocks/pubsub/pubsub-raw

Reply:
Ah you're right!